packages/issuer/lib/VcIssuer.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  encodeJsonAsURI,
+  ICredentialIssuerMetadataParametersV1_11,
+  ICredentialSuccessResponse,
+  IIssueCredentialRequest,
+  unsupported_credential_format,
+} from '@sphereon/openid4vci-common'
+
+import { v4 as uuidv4 } from 'uuid'
+
+export class VcIssuer {
+  _issuerMetadata: ICredentialIssuerMetadataParametersV1_11
+  _preAuthorizedCode?: string
+  _userPinRequired?: boolean
+  constructor(issuerMetadata: ICredentialIssuerMetadataParametersV1_11, preAuthorizedCode?: string, userPinRequired?: boolean) {
+    this._issuerMetadata = issuerMetadata
+    this._preAuthorizedCode = preAuthorizedCode
```

Comment:
Hmzz, not sure whether something which is request specific should be part of the issuer, which should be able to handle multiple request responses in the future

Reply:
removed, now create deeplink directly receives this as a param